gradle-suppressible-error-prone/src/main/java/com/palantir/gradle/suppressibleerrorprone/SuppressibleErrorPronePlugin.java

```diff
@@ -110,6 +110,12 @@ private void applyToJavaProject(Project project) {
                 });
             });
         }
+
+        project.getTasks().register("compileAllErrorProne", Task.class, compileAll -> {
+            compileAll.dependsOn(project.getTasks().withType(JavaCompile.class).matching(javaCompile -> {
+                return errorProneOptionsFor(javaCompile).getEnabled().get();
```

Comment:
this is a pitfall, if enablement changes at some later point, you'll never get these in your filtered `TaskCollection`.

I *think* you might want to wrap it in a provider perhaps? such that it's evaluated when `dependsOn` needs it, vs in configuration?

Reply:
I thought `matching` was lazy? It really hope it is at least. I guess my test could be working because `compileAllErrorProne` has not be realised by the time we disable it...

Reply:
hmm, you *may* be okay, perhaps I'm overfitting due to what I had to deal with in an internal project. It is lazy, it just depends when you that collection is iterated upon, so I imagine it *should* be fine as long as you don't try to do anything with it during configuration?

Reply:
I'm just going to add a provider just in case